pkg/kp/healthmanager.go

```diff
@@ -0,0 +1,340 @@
+package kp
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/square/p2/Godeps/_workspace/src/github.com/Sirupsen/logrus"
+	"github.com/square/p2/Godeps/_workspace/src/github.com/hashicorp/consul/api"
+
+	"github.com/square/p2/pkg/logging"
+	"github.com/square/p2/pkg/util/stream"
+)
+
+type consulSimpleHealthManager struct {
+	store  consulStore
+	logger logging.Logger
+}
+
+type consulSimpleHealthUpdater struct {
+	store      consulStore
+	logger     logging.Logger
+	lastCheck  time.Time
+	lastStatus string
+}
+
+// newSimpleHealthManager creates a simple health manager that writes a service's health
+// status to Consul every 15 seconds or whenever its status changes.
+func (c consulStore) newSimpleHealthManager(
+	_node string,
+	logger logging.Logger,
+) HealthManager {
+	return &consulSimpleHealthManager{
+		store:  c,
+		logger: logger,
+	}
+}
+
+// NewUpdater implements the HealthManager interface.
+func (m *consulSimpleHealthManager) NewUpdater(_service string) HealthUpdater {
```

Comment:
I'd have expected a truly unused param to be named `_`, which is why I was confused. If you're leaving them in for documentation-ish purposes, maybe an explanatory comment is in order.